selection/database_connector.py

```diff
@@ -43,6 +50,30 @@ def _prepare_query(self, query):
             elif "select" in query_statement or "SELECT" in query_statement:
                 return query_statement
 
+    def simulate_index(self, index):
+        self.simulated_indexes += 1
+
+        start_time = time.time()
+        result = self._simulate_index(index)
+        end_time = time.time()
+        self.index_simulation_duration += end_time - start_time
```

Comment:
I usually use and would prefer `self.index_simulation_duration += time.time() - start_time`.